chapter09/red/exercise3.pl

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use Regexp::Assemble;
+
+my ( $pattern_repo ) = @ARGV;
+if ( !defined $pattern_repo || !-r $pattern_repo ) {
+  print "Please supply a file containing line delimited regular expressions.\n";
+  exit 0;
+}
+
+my $regex = eval { new Regexp::Assemble( file => $pattern_repo )->re };
+die "Invalid regular expression: $@" if $@;
+
```

Comment:
This might die because your file can't be opened, although I think I forgot to check for that myself.